devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java

```diff
@@ -115,6 +118,17 @@ void generateCode(PathCollection sourceParents, Consumer<Path> sourceRegistrar,
                             Path serializedTestAppModelPath = BootstrapUtils
                                     .getSerializedTestAppModelPath(Path.of(mavenProject().getBuild().getDirectory()));
                             ApplicationModelSerializer.serialize(appModel, serializedTestAppModelPath);
+
+                            Properties properties = mavenProject().getProperties();
+                            String argLine = properties.getProperty("argLine", "");
+                            properties.setProperty("argLine", argLine +
+                                    " -D" + BootstrapConstants.SERIALIZED_TEST_APP_MODEL + "=" + serializedTestAppModelPath);
```

Comment:
Is this still necessary?

Reply:
Oh, now I see what you mean. We'd need to apply further changes to make it more acceptable for the Maven case.
The way it works now, there is a check whether the workspace has changed since the `ApplicationModel` was serialized to disk. This check will be bypassed, if we pass a system property. Meaning if I build a project, then go update the POM and re-run the tests, the changes to the POM would be ignored if we pass a system property.

Reply:
That's assuming `generate-code-tests` goal is configured. It has not been a requirement.
The main purpose of this goal is to generate source code for certain extensions. Most of the extensions don't actually need it.
The test `ApplicationModel` can be serialized on the first test run w/o this goal being configured though.

Reply:
I guess we could say if the goal is configured the model is re-generated on every run though.